tests/Routing/RouteTest.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace Test\Routing;
+
+use Test\BaseTestCase;
+use Zap\Routing\ControllerDispatcher;
+use Zap\Routing\Route;
+
+class RouteTest extends BaseTestCase
+{
+    /**
+     * @covers Zap\Routing\Route::define
+     */
+    public function testDefine()
+    {
+        $this->assertInstanceOf(Route::class, Route::define('/some/uri'));
+    }
+
+    /**
+     * @covers Zap\Routing\Route::getUri
+     */
+    public function testGetUri()
+    {
+        $this->assertInternalType('string', Route::define('/some/uri')->getUri());
+    }
+
+    /**
+     * @covers Zap\Routing\Route::getMethod
+     */
+    public function testGetMethod()
+    {
+        $this->assertInternalType('string', Route::define('/some/uri')->getMethod());
+    }
+
+    /**
+     * @covers Zap\Routing\Route::setMethod
+     */
+    public function testSetMethod()
+    {
+        $this->assertInstanceOf(Route::class,  Route::define('/some/uri')->setMethod('GET'));
```

Comment:
In all those set tests I would also add an assertion making sure the getter is retuning the updated value.